src/Forms/JoinSubtitles.cs

```diff
@@ -69,162 +70,137 @@ private void listViewParts_DragEnter(object sender, DragEventArgs e)
             SubtitleFormat lastFormat = null;
-            var subtitles = new List<Subtitle>();
-            for (int k = 0; k < _fileNamesToJoin.Count; k++)
+
+            for (int i = 0; i < _fileNamesToJoin.Count; i++)
             {
-                string fileName = _fileNamesToJoin[k];
+                string fileName = _fileNamesToJoin[i];
                 try
                 {
-                    var sub = new Subtitle();
-                    Encoding encoding;
-                    var format = sub.LoadSubtitle(fileName, out encoding, null);
+                    SubtitleFormat format = GetSubtitleFormat(fileName, out Subtitle sub);
 
+                    // unable to find format stop the operation
                     if (format == null)
                     {
-                        var ebu = new Ebu();
-                        if (ebu.IsMine(null, fileName))
-                        {
-                            ebu.LoadSubtitle(sub, null, fileName);
-                            format = ebu;
-                        }
-                    }
-                    if (format == null)
-                    {
-                        var pac = new Pac();
-                        if (pac.IsMine(null, fileName))
-                        {
-                            pac.LoadSubtitle(sub, null, fileName);
-                            format = pac;
-                        }
-                    }
-                    if (format == null)
-                    {
-                        var cavena890 = new Cavena890();
-                        if (cavena890.IsMine(null, fileName))
-                        {
-                            cavena890.LoadSubtitle(sub, null, fileName);
-                            format = cavena890;
-                        }
-                    }
-
-                    if (format == null)
-                    {
-                        for (int j = k; j < _fileNamesToJoin.Count; j++)
-                        {
-                            _fileNamesToJoin.RemoveAt(j);
-                        }
-
-                        MessageBox.Show("Unkown subtitle format: " + fileName);
+                        CancelMergeOperation(i, fileName);
                         return;
                     }
-                    if (sub.Header != null)
+
+                    // use last subtitle header if available
+                    if (!string.IsNullOrEmpty(sub.Header))
                     {
                         header = sub.Header;
                     }
 
-                    if (lastFormat == null || lastFormat.FriendlyName == format.FriendlyName)
+                    // set last-format if not set yet
+                    if (lastFormat == null)
                     {
                         lastFormat = format;
                     }
                     else
                     {
-                        lastFormat = new SubRip(); // default subtitle format

```

Comment:
Unnecessary instance of subrip were being created every time previous format isn't same as current format in loop